metricbeat/module/ceph/cluster_status/data.go

```diff
@@ -0,0 +1,132 @@
+package cluster_status
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type PgState struct {
+	Count     int64  `json:"count"`
+	StateName string `json:"state_name"`
+}
+
+type Pgmap struct {
+	AvailByte int64 `json:"bytes_avail"`
+	TotalByte int64 `json:"bytes_total"`
+	UsedByte  int64 `json:"bytes_used"`
+	DataByte  int64 `json:"data_bytes"`
+
+	DegradedObjs  int64   `json:"degraded_objects"`
+	DegradedRatio float64 `json:"degraded_ratio"`
+	DegradedTotal int64   `json:"degraded_total"`
+
+	MisplacedObjs  int64   `json:"misplaced_objects"`
+	MisplacedRatio float64 `json:"misplaced_ratio"`
+	MisplacedTotal int64   `json:"misplaced_total"`
+
+	ReadByteSec  int64 `json:"read_bytes_sec"`
+	ReadOpSec    int64 `json:"read_op_per_sec"`
+	WriteByteSec int64 `json:"write_bytes_sec"`
+	WriteOpSec   int64 `json:"write_op_per_sec"`
+	Version      int64 `json:"version"`
+
+	PgNum    int64     `json:"num_pgs"`
+	PgStates []PgState `json:"pgs_by_state"`
+}
+
+type Osdmap struct {
+	Epoch      int64 `json:"epoch"`
+	Full       bool  `json:"full"`
+	Nearfull   bool  `json:"nearfull"`
+	OsdNum     int64 `json:"num_osds"`
+	UpOsds     int64 `json:"num_up_osds"`
+	InOsds     int64 `json:"num_in_osds"`
+	RemapedPgs int64 `json:"num_remapped_pgs"`
+}
+
+type Osdmap_ struct {
+	Osdmap Osdmap `json:"osdmap"`
+}
+
+type Output struct {
+	Pgmap  Pgmap   `json:"pgmap"`
+	Osdmap Osdmap_ `json:"osdmap"`
+}
+
+type HealthRequest struct {
+	Status string `json:"status"`
+	Output Output `json:"output"`
+}
+
+func eventsMapping(content []byte) []common.MapStr {
+	var d HealthRequest
+	err := json.Unmarshal(content, &d)
+	if err != nil {
+		logp.Err("Error: ", err)
```

Comment:
I'm thinking it would be better to return the error from this function. The caller should then check if an error is returned and take the appropriate action (which in this case should be return the error from `Fetch`).

Reply:
Hi,  I return the error in the new commit a65dd21, please review it , thanks.